compiler/src/dotty/tools/dotc/inlines/Inliner.scala

```diff
@@ -768,6 +790,7 @@ class Inliner(val call: tpd.Tree)(using Context):
     override def typedSelect(tree: untpd.Select, pt: Type)(using Context): Tree = {
       val locked = ctx.typerState.ownedVars
       val qual1 = typed(tree.qualifier, shallowSelectionProto(tree.name, pt, this, tree.nameSpan))
+      selectionType(tree, qual1) // side-effect
```

Comment:
It would be good to be more specific what side effect (error messages?, anything else?)